src/route-handlers/fetch-workflow-query-types/fetch-workflow-query-types.ts

```diff
@@ -0,0 +1,61 @@
+import { NextResponse, type NextRequest } from 'next/server';
+
+import decodeUrlParams from '@/utils/decode-url-params';
+import * as grpcClient from '@/utils/grpc/grpc-client';
+import { getHTTPStatusCode, GRPCError } from '@/utils/grpc/grpc-error';
+import logger, { type RouteHandlerErrorPayload } from '@/utils/logger';
+
+import { type RequestParams } from './fetch-workflow-query-types.types';
+import parseErrorMessageForQueryTypes from './helpers/parse-error-message-for-query-types';
+import { queryTypesDataSchema } from './schemas/query-types-data-schema';
+
+export default async function fetchWorkflowQueryTypes(
+  _: NextRequest,
+  requestParams: RequestParams
+) {
+  const decodedParams = decodeUrlParams(requestParams.params);
```

Comment:
Nit: adding queryParams schema can help mark queryParams as required and provide better error message to the user than depending on the backend response

Reply:
We're not using any query params in this handler, we're using only URL params (and those are already required for the route to be resolved)